### PR TITLE
Move Satellite into Telescope

### PR DESCRIPTION
**Pull request by menghif**
## Issue This PR Addresses

Fixes: #2677

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This PR moves the Satellite project into the Telescope monorepo.

I have added a contributors section in  satellite's `package.json` to give credit to all previous contributors since the git history will not carry over. Once this merges we will be able to archive the original satellite repo. **UPDATE**: we used `git subtree` so the git history is not lost!

Satellite will be published from Github Actions with pnpm here:

https://github.com/Seneca-CDOT/telescope/blob/50ba194a5c61ba8bfc2b439dbca13264145ecd4e/.github/workflows/release.yml#L50

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -1,9 +1,6 @@
 {
   "extends": ["config:base"],
-  "packageRules": [
-    { "matchPackagePatterns": ["@senecacdot/satellite"], "rangeStrategy": "bump" },
```
Comment (humphd) [inline .github/renovate.json:4]: We might still need this, I don't think moving it into the repo will change how we do updates.

File: .prettierrc [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "endOfLine": "lf",
   "htmlWhitespaceSensitivity": "css",
   "insertPragma": false,
+  "jsxBracketSameLine": false,
```
Comment (humphd) [inline .prettierrc:8]: Why this change here?

**Review by humphd — CHANGES_REQUESTED** [review]
I'm so excited to see this!

A couple small things

**Comment by humphd** [comment]
NOTE: please don't squash this, we are going to keep all the commit history from Satellite here.

File: .prettierrc (hunk 2 above)
Comment (menghif) [reply .prettierrc]: This was in Satellite but not in Telesope. I don't think we actually need it.

**Review by cindyledev — APPROVED** [review]
I'm just gonna casually say that I still cannot believe this is possible

**Comment by menghif** [comment]
> @menghif are we agree on using sub-tree ?

Yes, so we get to keep the whole git history.

> I'm just gonna casually say that I still cannot believe this is possible

Same here. Thanks to @DukeManh for teaching us this!

**Comment by tpmai22** [comment]
@cindyledev Technology they said is easy

**Comment by DukeManh** [comment]
```diff
--  Thanks to @DukeManh for teaching us this!
++ Thanks to @DukeManh for suggesting this!
```

@menghif you are most welcome.

**Comment by cindyledev** [comment]
> ```diff
> --  Thanks to @DukeManh for teaching us this!
> ++ Thanks to @DukeManh for suggesting this!
> ```
> 
> @menghif you are most welcome.

wtf I just learned how to do diff code blocks from this. I literally thought you had to do some weird jsx magic to accomplish it. Meanwhile it was just `diff`

**Comment by DukeManh** [comment]
@menghif, any Turbo repo need to publish Satellite?

**Comment by menghif** [comment]
> @menghif, any Turbo repo config needed to publish Satellite?

Yes but I'll make the changes in a followup PR so it's easier to review.

**Comment by cindyledev** [comment]
Look at this! It even pulled in all the contributors of Satellite into this. git magic

2022-03-08 10_52_55-Window

File: src/satellite/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Define the env variables our code expects
```
Comment (humphd) [inline src/satellite/jest.config.js:1]: We need a follow-up to wire this project into our Jest config too (e.g., define it in our list of projects).  @menghif same question: can you fix in a follow-up or file?

**Review by humphd — APPROVED** [review]
Looks good, just a few things for follow-up.

File: src/satellite/jest.config.js (hunk 3 above)
Comment (menghif) [reply src/satellite/jest.config.js:1]: Will file it

**Comment by tpmai22** [comment]
@menghif  @DukeManh Great work !
